shell/browser/ui/win/notify_icon_host.cc

```diff
@@ -142,14 +143,14 @@ LRESULT CALLBACK NotifyIconHost::WndProc(HWND hwnd,
     }
     return TRUE;
   } else if (message == kNotifyIconMessage) {
-    NotifyIcon* win_icon = NULL;
+    base::WeakPtr<NotifyIcon> win_icon = NULL;
 
     // Find the selected status icon.
     for (NotifyIcons::const_iterator i(notify_icons_.begin());
          i != notify_icons_.end(); ++i) {
       auto* current_win_icon = static_cast<NotifyIcon*>(*i);
       if (current_win_icon->icon_id() == wparam) {
-        win_icon = current_win_icon;
+        win_icon = current_win_icon->GetWeakPtr();
```

Comment:
keep this loop as it was before and call `->GetWeakPtr()` after finding the icon. That way we call `GetWeakPtr` only once instead of _n_ times.